### ISSUE TITLE
Some Suggestions and one bug (?)

### ISSUE THREAD

**Issue by dsdoermann** [robindijkhof/overleaf-textarea#50]
I want to again express my thanks for this tool.  I can see it evolving and it is great.

Previously I had some scripts that processed the tex file before passing it to detex.  I don't know how your code works, but here are a few things I found useful for avoiding Grammarly errors when there were none.

1) For cite and refs, before detex I prepend a token "CITE" (CITE\cite{}), then when the latex is removed, the text still shows "in CITE" and no Grammarly error is generated.

2) I remove all double spaces since they don't matter in latex. (Again I don't know how your code works)

3) Im getting a situation when I have a comma after a \cite, showing up a period in the text.  "Multi-PIE dataset \cite{multipie}, which" shows up as  " Multi-PIE dataset. which "  in the text panel, which generates a Grammarly error

4) It seems there is  a problem with the text area if there are two citations in one line?

The loss function used is the same as in \cite{DBLP:journals/corr/abs-1710-00925}. The loss function has two components, the bin classification component and the regression component. Bin classification uses stable softmax and cross-entropy, which helps the network predict the pose's neighborhood. N. Ruiz et al. \cite{DBLP:journals/corr/abs-1710-00925} calculated three entropy losses, one for each angle, and backpropagated it through the network to improve learning.  They computed the expectation of each output angle from the binned output to obtain the fine-grained prediction. The regression loss was calculated using the Mean Square Error loss to improve the fine-grained prediction. The alpha value is a hyperparameter used for the regression loss function, and the weight of the classification loss is constant.

shows up  in textarea as 

The loss function used is the same as in calculated three entropy losses, one for each angle, and backpropagated it through the network to improve learning.  They computed the expectation of each output angle from the binned output to obtain the fine-grained prediction. The regression loss was calculated using the Mean Square Error loss to improve the fine-grained prediction. The alpha value is a hyperparameter used for the regression loss function, and the weight of the classification loss is constant.

So everything is missing between the two citations.

**Comment by robindijkhof** [comment]
Thanks, I'll check it out. Your first point can be achieved by using custom filters. Something like this would work:

filter: /(\\cite{)(.*?)(})/g
value: CITE

**Comment by dsdoermann** [comment]
Thank you so much!  I just realized that there are custom filters.  I'll try that next time.  I'm not a great programmer anymore, but I think I can handle some regex!

I also see a sync scroll now, but I don't remember updating.  Do I need to update to get the latest version manually?  May it be nice to have the version or version date on the plugin on the toolbar?



Dave


---------------------------
David Doermann
Professor, SUNY Empire Innovation
Professor, Department of Computer Science and Engineering
University at Buffalo

email: ***@***.***
website: cse.buffalo.edu/~doermann<http://cse.buffalo.edu/~doermann>
phone: 716-645-1557 (office), 410-493-9043 (mobile)

[footer_logo_03]

**Comment by dsdoermann** [comment]
Just FYI, I needed double slash to get it to work.

Thanks!


---------------------------
David Doermann
Professor, SUNY Empire Innovation
Professor, Department of Computer Science and Engineering
University at Buffalo

email: ***@***.***
website: cse.buffalo.edu/~doermann<http://cse.buffalo.edu/~doermann>
phone: 716-645-1557 (office), 410-493-9043 (mobile)

[footer_logo_03]

**Comment by dsdoermann** [comment]
And somehow that fixes the "bug" I suggested.


---------------------------
David Doermann
Professor, SUNY Empire Innovation
Professor, Department of Computer Science and Engineering
University at Buffalo

email: ***@***.***
website: cse.buffalo.edu/~doermann<http://cse.buffalo.edu/~doermann>
phone: 716-645-1557 (office), 410-493-9043 (mobile)

[footer_logo_03]
